admin-panel/src/main/java/net/cryptic_game/backend/admin/endpoints/website/blog/BlogEndpoints.java

```diff
@@ -0,0 +1,101 @@
+import net.cryptic_game.backend.base.api.data.ApiResponse;
+import net.cryptic_game.backend.base.api.data.ApiType;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+
+import java.time.OffsetDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@ApiEndpointCollection(id = "website/blog", description = "manages blog", type = ApiType.REST, authenticator = AdminPanelAuthenticator.class)
+public final class BlogEndpoints {
+
+    private final BlogEntryRepository blogEntryRepository;
+
+    @ApiEndpoint(id = "list")
+    public ApiResponse list(@ApiParameter(id = "lang", required = false) final String lang,
+                            @ApiParameter(id = "page") final int page,
+                            @ApiParameter(id = "page_size") final int pageSize) {
+        if (lang == null) {
+            return new ApiResponse(HttpResponseStatus.OK, this.shortContent(
+                    this.blogEntryRepository.findAllByOrderByCreationDesc(PageRequest.of(page, pageSize))));
+        }
+        return new ApiResponse(HttpResponseStatus.OK, this.shortContent(
+                this.blogEntryRepository.findAllByLanguageOrderByCreationDesc(lang, PageRequest.of(page, pageSize))));
+    }
+
+    @ApiEndpoint(id = "get")
+    public ApiResponse getBlogEntry(@ApiParameter(id = "id") final UUID id) {
+        final Optional<BlogEntry> entry = this.blogEntryRepository.findById(id);
+        if (entry.isEmpty()) {
+            return new ApiResponse(HttpResponseStatus.NOT_FOUND, "NO_SUCH_ELEMENT");
+        }
+        return new ApiResponse(HttpResponseStatus.OK, entry.get());
+    }
+
+    @ApiEndpoint(id = "add", authentication = Permission.BLOG_MANAGEMENT)
+    public ApiResponse add(@ApiParameter(id = "title") final String title,
+                           @ApiParameter(id = "content") final String content,
+                           @ApiParameter(id = "lang") final String lang) {
+        if (WebsiteUtils.checkXxs(title) || WebsiteUtils.checkXxs(content)) {
+            return new ApiResponse(HttpResponseStatus.FORBIDDEN, "NO_HTML_TAGS_ALLOWED");
+        }
+        if (content.length() > WebsiteUtils.MAX_CONTENT_LENGTH) {
+            return new ApiResponse(HttpResponseStatus.FORBIDDEN, "CONTENT_TOO_LONG");
+        }
+        return new ApiResponse(HttpResponseStatus.OK, this.blogEntryRepository.save(new BlogEntry(OffsetDateTime.now(), null, title, lang, content)));
+    }
+
+    @ApiEndpoint(id = "update", authentication = Permission.BLOG_MANAGEMENT)
+    public ApiResponse update(@ApiParameter(id = "id") final UUID id,
+                              @ApiParameter(id = "title") final String title,
+                              @ApiParameter(id = "content") final String content) {
+        if (WebsiteUtils.checkXxs(title) || WebsiteUtils.checkXxs(content)) {
+            return new ApiResponse(HttpResponseStatus.FORBIDDEN, "NO_HTML_TAGS_ALLOWED");
+        }
```

Comment:
May use [baeldung.com/spring-prevent-xss#4-stripping-xss-from-the-request-body](https://www.baeldung.com/spring-prevent-xss#4-stripping-xss-from-the-request-body) cause of better security.